tests/test_agent_manager.py

```diff
@@ -1278,6 +1253,99 @@ async def test_dont_start_paused_agent(server, client, environment, caplog) -> N
     assert "took too long to start" not in caplog.text
 
 
+async def test_wait_for_agent_map_update(server, client, environment) -> None:
```

Comment:
I verified that both these tests fail on master. They verify the core change in this PR.